File: Ejercicios Java/Condicionales.java
import java.util.Scanner;

public class Condicionales {
    @SuppressWarnings("ConvertToTryWithResources")
    public static void main(String[] args) {
        // Crear el scanner para leer datos
        Scanner scanner = new Scanner(System.in);
        
        // Pedir al usuario que seleccione una figura para calcular su área
        System.out.println("Seleccione la figura a calcular su área:");
        System.out.println("1 para rombo");
        System.out.println("2 para rectángulo");
        System.out.println("3 para cuadrado");
        System.out.println("4 para trapecio");
        
        // Leer la opción seleccionada
        int figura = scanner.nextInt();
        double area;

        // Ejecutar el cálculo de área según la figura seleccionada
        switch (figura) {
            case 1 -> {
                // Rombo
                System.out.print("Ingresa el valor de la diagonal mayor: ");
                double Dmayor = scanner.nextDouble();
                System.out.print("Ingresa el valor de la diagonal menor: ");
                double Dmenor = scanner.nextDouble();
                area = (Dmayor * Dmenor) / 2;
                System.out.println("El área del rombo es: " + area);
            }

            case 2 -> {
                // Rectángulo
                System.out.print("Ingresa el valor del largo del rectángulo: ");
                double largo = scanner.nextDouble();
                System.out.print("Ingresa el valor del ancho del rectángulo: ");
                double ancho = scanner.nextDouble();
                area = largo * ancho;
                System.out.println("El área del rectángulo es: " + area);
            }

            case 3 -> {
                // Cuadrado
                System.out.print("Ingresa el valor del lado del cuadrado: ");
                double lado = scanner.nextDouble();
                area = lado * lado;
                System.out.println("El área del cuadrado es: " + area);
            }

            case 4 -> {
                // Trapecio
                System.out.print("Ingresa el valor de la base mayor: ");
                double Bmayor = scanner.nextDouble();
                System.out.print("Ingresa el valor de la base menor: ");
                double Bmenor = scanner.nextDouble();
                System.out.print("Ingresa la altura del trapecio: ");
                double H = scanner.nextDouble();
                area = ((Bmayor + Bmenor) * H) / 2;
                System.out.println("El área del trapecio es: " + area);
            }

            default -> System.out.println("Opción no válida.");
        }

        // Cerrar el scanner al final
        scanner.close();
    }
}

File: Ejercicios_Rubi/Condicionales.rb
puts "selecione la figura a calcular su area: "
puts "1 para rombo"
puts "2 para rectangulo"
puts "3 para cuadrado"
puts "4 para trapecio"
figura= gets.chomp 
Pi=3.1416
Const= 2
case figura #En lugar de match en rubi se utiliza case 
when '1' # Rombo
  puts "ingresa el valor de la diagonal mayor: "
  dmayor = gets.to_f
  puts "ingresa el valor de la diagonal menor: "
  dmenor = gets.to_f
  area = (dmayor*dmenor)/Const
  puts "El area del rombo es: #{area}"
when '2' # Rectangulo
  puts "ingresa el valor del largo del rectangulo: "
  largo = gets.to_f
  puts "ingresa el valor del ancho del rectangulo: "
  ancho = gets.to_f
  area = (largo*ancho)
  puts "El area del rectangulo es: #{area}"
when '3' # Cuadrado
  puts "ingresa el valor del lado del cuadrado: "
  lado = gets.to_f
  area = (lado*lado)
  puts "El area del cuadrado es: #{area}"         
when '4' # Trapecio
  puts "ingresa el valor de la base mayor: "
  bmayor = gets.to_f
  puts "ingresa el valor de la base menor: "
  bmenor = gets.to_f
  puts "ingrese la altura del trapecio: "
  h = gets.to_f
  area = ((bmayor+bmenor)*h)/Const
  puts "El area del trapecio es: #{area}"
else # else para crear una condicion en la que se cierre case
  puts "Opción no válida. Por favor, elija una opción entre 1 y 4."
end